buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -289,47 +362,130 @@ private void waitForProcessToExit(ProcessHandle processHandle) {
-
-    private File getConfPathRepo() {
-        return new File(workingDir, "repo");
+    private void createWorkingDir(Path distroExtractDir) {
+        try {
+            syncWithLinks(distroExtractDir, workingDir);
+            Files.createDirectories(confPathRepo);
+            Files.createDirectories(confPathData);
+            Files.createDirectories(workingDir.resolve("sharedData"));
+            Files.createDirectories(confPathLogs);
+            Files.createDirectories(tmpDir);
+        } catch (IOException e) {
+            throw new UncheckedIOException("Failed to create working directory: " + workingDir, e);
+        }
     }
 
-    private File getConfPathLogs() {
-        return new File(workingDir, "logs");
+    public void syncWithLinks(Path sourceRoot, Path destinationRoot) throws IOException {
+        // There's some latency in Windows between when a cluster running here previously releases the files so we can
+        // can clean them up. Make sure we can run the same clusters in quick succession.
+        removeWithRetry(destinationRoot);
+        try (Stream<Path> stream = Files.walk(sourceRoot)) {
+            stream.forEach(source -> {
+                Path destination = destinationRoot.resolve(sourceRoot.relativize(source));
+                if (Files.isDirectory(source)) {
+                    try {
+                        Files.createDirectories(destination);
+                    } catch (IOException e) {
+                        throw new UncheckedIOException("Can't create directory " + destination.getParent(), e);
+                    }
+                } else {
+                    try {
+                        Files.createDirectories(destination.getParent());
+                    } catch (IOException e) {
+                        throw new UncheckedIOException("Can't create directory " + destination.getParent(), e);
+                    }
+                    try {
+                        Files.createLink(destination, source);
+                    } catch (IOException e) {
+                        // Note does not work for network drives, e.x. Vagrant
+                        throw new UncheckedIOException(
+                            "Failed to create hard link " + destination + " pointing to " + source, e
+                        );
+                    }
+                    try {
+                        setPermissionsReadOnly(source, destination);
+                    } catch (IOException e) {
+                        throw new UncheckedIOException(
+                            "Failed to set permissions " + destination + " pointing to " + source, e
+                        );
+                    }
+                }
+            });
+        }
     }
 
-    private File getStdoutFile() {
-        return new File(getConfPathLogs(), "es.stdout.log");
+    private void removeWithRetry(Path destinationRoot) {
+        if (Files.exists(destinationRoot)) {
+            for (int tries = 1; true; tries++) {
```

Comment:
I don't think we should ever have an infinite loop by design. There should be a max number of tries. I would also structure this as a `while (tries < maxTries)` (but see my comment before about why I don't think we should have this method at all).

Reply:
I had `tries <= maxTries` there and IDEA suggested it can be simplified to `true`, I wen't with it.  Why would you use a `while` over a `for` ? I'll add condition back so it doesn't seem like an infinite loop.